service/src/main/java/bio/terra/workspace/service/resource/model/ResourceLineageEntry.java

```diff
@@ -52,4 +52,10 @@ public boolean equals(Object o) {
     return sourceWorkspaceId.equals(entry.sourceWorkspaceId)
         && sourceResourceId.equals(entry.sourceResourceId);
   }
+
+  @Override
+  public int hashCode() {
```

Comment:
Why `assert false`, could you implement a real hashCode? (Or delete this method?)

Reply:
this  was  a complaint from spotbugs and this is its suggestions. because ResourceLineageEntry won't be put into a hashmap or hashtable, it assert false. I needed to have a hashcode because i have a  custom implementation of equals.

Reply:
ResourceLineageEntry might be put into hashmap or hashtable someday, seems safer to just implement hashCode